bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/api/BondHttpApi.java

```diff
@@ -0,0 +1,266 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bondhome.internal.api;
+
+import static org.openhab.binding.bondhome.internal.BondHomeBindingConstants.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.InputStreamContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.bondhome.internal.BondException;
+import org.openhab.binding.bondhome.internal.handler.BondBridgeHandler;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+
+/**
+ * {@link BondHttpApi} wraps the Bond REST API and provides various low
+ * level function to access the device api (not cloud api).
+ *
+ * @author Sara Geleskie Damiano - Initial contribution
+ */
+@NonNullByDefault
+public class BondHttpApi {
+    private final Logger logger = LoggerFactory.getLogger(BondHttpApi.class);
+    private final BondBridgeHandler bridgeHandler;
+    private final HttpClientFactory httpClientFactory;
+    private Gson gson = new Gson();
+
+    public BondHttpApi(BondBridgeHandler bridgeHandler, final HttpClientFactory httpClientFactory) {
+        this.bridgeHandler = bridgeHandler;
+        this.httpClientFactory = httpClientFactory;
+    }
+
+    /**
+     * Gets version information about the Bond bridge
+     *
+     * @return the {@link BondSysVersion}
+     * @throws IOException
+     */
+    @Nullable
+    public BondSysVersion getBridgeVersion() throws BondException {
+        String json = request("/v2/sys/version");
+        logger.trace("BondHome device info : {}", json);
+        try {
+            return gson.fromJson(json, BondSysVersion.class);
+        } catch (JsonParseException e) {
+            logger.debug("Could not parse sys/version JSON '{}'", json, e);
+            throw new BondException("@text/offline.comm-error.unparseable-response");
```

Comment:
You could consider adding another constructor to `BondException` also taking a `Throwable`. This could then be chained, so you wouldn't have to log the details here, but instead could let the caller `getCause()` and log:
```suggestion
            throw new BondException("@text/offline.comm-error.unparseable-response", e);
```

I haven't checked how much you would benefit from that, so it's only for consideration if it would be beneficial.

Reply:
OK.